src/client/dfuse/dfuse_main.c

```diff
@@ -428,87 +425,110 @@ main(int argc, char **argv)
 		}
 	}
 
+	if (cont_name && !pool_name) {
+		printf("Container name specified without pool\n");
+		D_GOTO(out_debug, rc = -DER_INVAL);
+	}
+
 	rc = daos_init();
 	if (rc != -DER_SUCCESS)
-		D_GOTO(out_debug, ret = rc);
+		D_GOTO(out_debug, rc);
 
 	DFUSE_TRA_ROOT(dfuse_info, "dfuse_info");
 
 	rc = dfuse_fs_init(dfuse_info, &fs_handle);
 	if (rc != 0)
-		D_GOTO(out_debug, ret = rc);
+		D_GOTO(out_debug, rc);
 
-	if (have_pool_label) {
-		rc = dfuse_pool_connect_by_label(fs_handle,
-						 dfuse_info->di_pool,
-						 &dfp);
-		if (rc != 0) {
-			printf("Failed to connect to pool (%d) %s\n",
-				rc, strerror(rc));
-			D_GOTO(out_dfs, ret = daos_errno2der(rc));
+	/* Firsly check for attributes on the path.  If this option is set then
+	 * it is expected to work.
+	 */
+	if (path) {
+		if (pool_name) {
+			printf("Pool specified multiple ways\n");
+			D_GOTO(out_daos, rc = -DER_INVAL);
```

Comment:
Are this extra cart errors expected?
```
$ dfuse -m dfuse --pool $pool1 --cont $cont1 --path uns
Pool specified multiple ways
crt  ERR  src/cart/crt_init.c:593 crt_finalize() cannot finalize, current ctx_num(1).
crt  ERR  src/cart/crt_init.c:647 crt_finalize() failed, rc: -1001.
client ERR  src/client/api/event.c:130 daos_eq_lib_fini() failed to shutdown crt: DER_NO_PERM(-1001): 'Operation not permitted'
client ERR  src/client/api/init.c:250 daos_fini() failed to finalize eq: DER_NO_PERM(-1001): 'Operation not permitted'
Exiting -1003 DER_INVAL
```

Reply:
Yes, for now.  It's because there are leaked references on shutdown which is something I'm working on elsewhere.  Whilst this isn't correct it's the same as for other error paths in this function.